pkg/sqlite/load.go

```diff
@@ -104,8 +110,85 @@ func (s *sqlLoader) addOperatorBundle(tx *sql.Tx, bundle *registry.Bundle) error
 	if err != nil {
 		return err
 	}
+	substitutesFor, err := bundle.SubstitutesFor()
+	if err != nil {
+		return err
+	}
+
+	// Update the replaces that reference the substituted-for bundle
+	if substitutesFor != "" {
+		_, err = updateBundle.Exec(csvName, substitutesFor)
+		if err != nil {
+			return err
+		}
+	}
+
+	// If the substituted-for bundle substitutes for another bundle
+	// it should also be added to the skips of the substitutesFor bundle
+	for substitutesFor != "" {
+		skips = append(skips, substitutesFor)
+		substitutesFor, err = s.getBundleSubstitution(tx, substitutesFor)
+		if err != nil {
+			return err
+		}
+	}
+
+	// If a new bundle replaces a bundle that is substituted for
+	// (for example it was the previous head of the channel), change
+	// the replaces to the substituted-for bundle and add corresponding skips
+	if replaces != "" {
+		substitutesFor, err = s.getBundleThatSubstitutesFor(tx, replaces)
+		if err != nil {
+			return err
+		}
+		for substitutesFor != "" {
+			// add a skips reference to the substitutedfor bundles
+			skips = append(skips, replaces)
+			// update the replaces to a newer substitution
+			replaces = substitutesFor
+			// try to get the substitution of the substitution
+			substitutesFor, err = s.getBundleThatSubstitutesFor(tx, replaces)
+			if err != nil {
+				return err
+			}
+		}
+	}
```

Comment:
IIUC, it seems like altering the graph here makes `addOperatorBundle` non-commutative; i.e. order of an addition matters -- which is something we can't have.

I think we want to do this _after_ all of the bundles have been added to the database, so we have complete substitutes-for information.

Reply:
all the tests have shown me that it's still commutative. Let me know if you see something I don't

Reply:
the tests were a little difficult for me to grok. 

stepping through the code a second time, it looks like this works out for `replaces`.